subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1677,7 +1658,51 @@ static void le_advertising_report(struct pdu_data *pdu_data, u8_t *b,
 			dup_curr = 0;
 		}
 	}
-fill_report:
+
+	return false;
+}
+
+static void le_advertising_report(struct pdu_data *pdu_data, u8_t *b,
+				  struct net_buf *buf)
+{
+	const u8_t c_adv_type[] = { 0x00, 0x01, 0x03, 0xff, 0x04,
+				    0xff, 0x02 };
+	struct bt_hci_evt_le_advertising_report *sep;
+	struct pdu_adv *adv = (struct pdu_adv *)pdu_data;
+	struct bt_hci_evt_le_advertising_info *adv_info;
+	u8_t data_len;
+	u8_t info_len;
+	u8_t rssi;
+#if defined(CONFIG_BT_CONTROLLER_PRIVACY)
+	u8_t rl_idx, direct;
+#endif /* CONFIG_BT_CONTROLLER_PRIVACY */
+	u8_t *prssi;
+
+	if (!(event_mask & BT_EVT_MASK_LE_META_EVENT)) {
+		return;
+	}
+
+#if defined(CONFIG_BT_CONTROLLER_PRIVACY)
+	rl_idx = b[offsetof(struct radio_pdu_node_rx, pdu_data) +
```

Comment:
move this closer to locality of reference, that way, it is not needed exiting at line 1693.